debezium-embedded/src/main/java/io/debezium/embedded/spi/OffsetCommitPolicy.java

```diff
@@ -18,7 +18,7 @@
  * @author Randall Hauch
  */
 @FunctionalInterface
-public interface OffsetCommitPolicy {
+public interface OffsetCommitPolicy extends io.debezium.engine.spi.OffsetCommitPolicy {
```

Comment:
Why do we need this internal interface?

Reply:
The same question applies for the inner interfaces in `EmbeddedEngine`. Why do we have these pairs of public and internal interface now?

Reply:
@gunnarmorling Well, I know that we have not declared `EmbeddedEngine` to be a public API but still people are using it, so everything I did ensures full backward compatibility. My intent is to mark these "internal" interfaces as deprecated and remove them in future releases.
If we drop them right away the users code will not be compilable.

Reply:
Ok, I see. So they should be marked with `@Deprecated` then. On backwards compat, I don't think that simply moving the types would be a breaking change strictly speaking. core pulls in api, so a user's code would compile just as before. The one issue to take care of would be the split package problem though, when only moving parts of a package.

Reply:
@gunnarmorling It is not just move among modules but in package name as well, hence my desire to keep the old ones.

Reply:
@gunnarmorling Deprecated annotations added